src/codeu/chat/client/LoginDialog.java

```diff
@@ -0,0 +1,125 @@
+package codeu.chat.client;
+
+import java.awt.*;
+import java.awt.event.*;
+import javax.swing.*;
+import javax.swing.border.*;
+
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+import codeu.chat.common.BasicView;
+import codeu.chat.common.Conversation;
+import codeu.chat.common.ConversationSummary;
+import codeu.chat.common.LogicalView;
+import codeu.chat.common.Message;
+import codeu.chat.common.NetworkCode;
+import codeu.chat.common.User;
+import codeu.chat.util.Logger;
+import codeu.chat.util.Serializers;
+import codeu.chat.util.Time;
+import codeu.chat.util.Uuid;
+import codeu.chat.util.connections.Connection;
+import codeu.chat.util.connections.ConnectionSource;
+import codeu.chat.client.simplegui.UserPanel;
+
+public class LoginDialog extends JDialog {
+ 
+    private JTextField tfUsername;
+    private JPasswordField pfPassword;
+    private JLabel lbUsername;
+    private JLabel lbPassword;
+    private JButton btnLogin;
+    private JButton btnCancel;
+    private boolean succeeded;
+ 
+    public LoginDialog(Frame parent) {
+        super(parent, "Login", true);
+        //
+        JPanel panel = new JPanel(new GridBagLayout());
+        GridBagConstraints cs = new GridBagConstraints();
+ 
+        cs.fill = GridBagConstraints.HORIZONTAL;
+ 
+        lbUsername = new JLabel("Username: ");
+        cs.gridx = 0;
+        cs.gridy = 0;
+        cs.gridwidth = 1;
+        panel.add(lbUsername, cs);
+ 
+        tfUsername = new JTextField(20);
+        cs.gridx = 1;
+        cs.gridy = 0;
+        cs.gridwidth = 2;
+        panel.add(tfUsername, cs);
+ 
+        lbPassword = new JLabel("Password: ");
+        cs.gridx = 0;
+        cs.gridy = 1;
+        cs.gridwidth = 1;
+        panel.add(lbPassword, cs);
+ 
+        pfPassword = new JPasswordField(20);
+        cs.gridx = 1;
+        cs.gridy = 1;
+        cs.gridwidth = 2;
+        panel.add(pfPassword, cs);
+        panel.setBorder(new LineBorder(Color.GRAY));
+ 
+        btnLogin = new JButton("Login");
+ 
+        btnLogin.addActionListener(new ActionListener() {
+ 
+            public void actionPerformed(ActionEvent e) {
+                if (Login.authenticate(getUsername(), getPassword())) {
```

Comment:
I'm confused what side-effect this code is supposed to produce. This doesn't look like it really does a login, it just validates a username / password pair? Where is the result of that validation consumed? Mostly because I don't think succeeded does anything, is it just implicit in the panel being disposed that the login flow continues?